pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,51 @@
+package ipsec
+
+import (
+	"sync"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode, recordedFlavor string = "", ""
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
+	registerMetrics.Do(func() {
+		ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name: "cluster_network_ipsec_state",
+			Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+		}, []string{"mode", "api_flavor"})
+		legacyregistry.MustRegister(ipsecStateGauge)
+	})
+}
+
+func UpdateIpsecTelemetry(ipsecConfig *operv1.IPsecConfig) {
+	var flavor, mode string
+	klog.V(5).Infof("IPsec Telemetry: %v", ipsecConfig)
+	RegisterIpsecMetrics()
+	if ipsecConfig == nil {
+		mode = string(operv1.IPsecModeDisabled)
+		flavor = "4.14"
```

Comment:
What is the `flavor` label about? Is it the OCP version?

Reply:
Yes - the API has changed between OCP 4.14 and OCP 4.15

Reply:
I see no reason to have it as a label since the cluster version is already available in Telemeter.

Reply:
I probably misled you in the previous response. There are 2 flavors to the API according to the version in which the API was invoked, but for backward compatibility we keep the API version during an upgrade until the API is invoked again, so you can find an instance of the old API (4.14) in newer clusters. The reason to keep the API flavor is in order to understand when we can drop the support for the old API flavor, which is non-trivial.